shared/ssl.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/sh
+#
+# Get a signed certificate for this host.
+#
+# Uses OpenSSL and curl directly to generate a new CSR and get it signed by the
+# Puppet Server CA.
+#
+# Intended to be used in place of a full-blown puppet agent run that is solely
+# for getting SSL certificates onto the host.
+#
+# Files will be placed in the same default directory location and structure
+# that the puppet agent would put them, which is /etc/puppetlabs/puppet/ssl.
+#
+# Fails under the following conditions:
+# - CA host cannot be reached
+# - CA already has signed certificate for this host
+# - SSL files already exist on this host
+#
+# Arguments:
+#   $1  (Optional) Certname to use, defaults to $HOSTNAME
+#
+# Optional environment variables:
+#   WAITFORCERT            Number of seconds to wait for certificate to be
+#                          signed, defaults to 120
+#   PUPPETSERVER_HOSTNAME  Hostname of Puppet Server CA, defaults to "puppet"
+#   SSLDIR                 Root directory to write files to, defaults to
+#                          "/etc/puppetlabs/puppet/ssl"
+
+msg() {
+    echo "($0) $1"
+}
+
+error() {
+    msg "Error: $1"
+    exit 1
+}
+
+### Verify dependencies available
+! which openssl > /dev/null && error "openssl not found on PATH"
```

Comment:
There are systems that do not have `which` (e.g., Fedora and CentOS base containers) It might be more portable to use `type -p openssl > /dev/null` instead.

Reply:
I knew there was some issue with using which, was hoping somebody would bring it up :)

Thanks, I'll redo this part.